sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageImplUtils.java

```diff
@@ -389,10 +391,27 @@ public Map.Entry<String, String> next() {
                 throw new NoSuchElementException();
             }
 
-            int nextPosition = queryParameters.indexOf('=', position);
+            int nextPosition = position;
+            char c;
+            while (nextPosition < queryParametersLength) {
+                // Next position can either be '=' or '&' as a query parameter may not have a '=', ex 'key&key2=value'.
+                c = queryParameters.charAt(nextPosition);
+                if (c == '=') {
+                    break;
+                } else if (c == '&') {
+                    String key = queryParameters.substring(position, nextPosition);
 
-            if (nextPosition == -1) {
-                // Query parameters completed with a key only 'https://example.com?param'
+                    // Position is set to nextPosition + 1 to skip over the '&'
+                    position = nextPosition + 1;
+
+                    return new AbstractMap.SimpleImmutableEntry<>(key, "");
+                }
+
+                nextPosition++;
+            }
+
+            if (nextPosition == queryParametersLength) {
```

Comment:
Do we have tests for StorageImplUtils?

Reply:
Given this is a one-for-one copy of Core not yet